### ISSUE TITLE
[Bug] Not found samples causes deleting lines in .h2song file?

### ISSUE THREAD

**Issue by oddtime** [hydrogen-music/hydrogen#754]
Version: 1.0.0-beta1  (aug  14 2019)
Kernel: Linux 4.4.0-97-generic (i686)
Distribution: Ubuntu 16.04.3 LTS

----------------------

When opening an old file, program didn't find some instrument samples - that was expected as samples had been converted to flac -
 BUT instead of muting that instruments, it DELETED all layer information in .h2song file.
Example:

In new saved file:
...
   <instrumentComponent>
    <component_id>0</component_id>
    <gain>1</gain>
   </instrumentComponent>
...

old file was:
...
   <instrumentComponent>
    <component_id>0</component_id>
    <gain>1</gain>
    <layer>
     <filename>16282_ltibbits_splash_A_custom_10_high_vol.wav</filename>
     <min>0.66</min>
     <max>1</max>
     <gain>1</gain>
     <pitch>0</pitch>
    </layer>
    <layer>
     <filename>16284_ltibbits_splash_A_custom_10_med_vol.wav</filename>
     <min>0.33</min>
     <max>0.66</max>
     <gain>1.75</gain>
     <pitch>0</pitch>
    </layer>
    <layer>
     <filename>16283_ltibbits_splash_A_custom_10_low_vol.wav</filename>
     <min>0</min>
     <max>0.33</max>
     <gain>4.25</gain>
     <pitch>0</pitch>
    </layer>
   </instrumentComponent>
...

**Comment by trebmuh** [comment]
I edited your message because it wasn't showing properly. Please, double check with the github message writing "preview" tab next time, and use the "Insert code" button too.

About your issue, I remember reading the an issue identical to that a few days/weeks ago.

@theGreatWhiteShark I kind of remember you wrote on this issue I've read. Do you remember which one it is?

**Comment by oddtime** [comment]
Sorry, I will check next time

**Comment by theGreatWhiteShark** [comment]
Yes, it's quite similar to #622 where layers of a drumkit generated using an older version of H2 were deleted since they are now recognized as invalid.

But I think this issue is different. Hydrogen is perfectly right in identifying the layers as invalid. I do read it more like a quite plausible feature request to change the default behavior from deleting the layers into just deactivating them.

**Comment by trebmuh** [comment]
> I do read it more like a quite plausible feature request to change the default behavior from deleting the layers into just deactivating them.

From what I understand, I do agree.

**Comment by oddtime** [comment]
> About your issue, I remember reading an issue identical to that a few days/weeks ago.
@trebmuh  Maybe it was me on devel mailing list? ;)

Ok for the "feature request". I thought it as a bug since it didn't happen before.
Thanks

**Comment by trebmuh** [comment]
Your message was still not displaying correctly. Please, use the "preview" functionality.

> @trebmuh Maybe it was me on devel mailing list? ;)

No, I don't think so. It was here on github.
